### PR TITLE
Userstories: Implemented Exception Handling & added Unit Tests

### PR DESCRIPTION
**Pull request by monil-persistent**
- Updated Intg Tests according to the current project
- Added embedded mongo to unit tests
- Added application.properties for unit tests as the main's application.properties is dependent on the server, i.e. to maintain isolation of unit tests
- Added one dependency in build.gradle for embedded mongo DB
- Excluded integration test from tasks in build.gradle as it is failing while running rbms[build] because rbms server must be running for integration test
- updated all input/expected testcases.jsons to according to the current project
- small fixes in the controller
- formatted code  and organized imports of updated classes

### CODE REVIEW

**Comment by monil-persistent** [comment]
Please review the following pull request and suggest any updates if needed.

File: src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@ControllerAdvice
+public class RbmsExceptionController {
+    @ExceptionHandler(value = RbmsException.class)
+    public ResponseEntity<Object> exception(RbmsException exception) {
+//        log.info("Field is null", exception);
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java:17]: Please remove this Commented log line

File: src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        Map<String, Object> map = new HashMap<String, Object>();
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java:18]: Give a proper name to this map object

File: src/main/java/com/wf/stp/rbms/listener/service/intermediator/IntermediatorService.java [hunk 3]
```diff
@@ -17,10 +19,13 @@ public class IntermediatorService implements RuleService {
...
-                && upoIn.getPmtInf().getComWellsfargoEpdUpoAvroPmtInf().getCdtrAgt().getComWellsfargoEpdUpoAvroCdtrAgt().getFinInstnId2().getComWellsfargoEpdUpoAvroFinInstnId2().getBicfi() != null
-                && !upoIn.getPmtInf().getComWellsfargoEpdUpoAvroPmtInf().getCdtrAgt().getComWellsfargoEpdUpoAvroCdtrAgt().getFinInstnId2().getComWellsfargoEpdUpoAvroFinInstnId2().getBicfi().equals("WFBIUS1A")
-                && !upoIn.getPmtInf().getComWellsfargoEpdUpoAvroPmtInf().getDbtrAgt().getComWellsfargoEpdUpoAvroDbtrAgt().getFinInstnId2().getComWellsfargoEpdUpoAvroFinInstnId2().getBicfi().equals("WFBIUS1A")) {
-            return true;
+                && upoIn.getPmtInf().getComWellsfargoEpdUpoAvroPmtInf().getCdtrAgt().getComWellsfargoEpdUpoAvroCdtrAgt().getFinInstnId2().getComWellsfargoEpdUpoAvroFinInstnId2().getBicfi() != null) {
+            if (!upoIn.getPmtInf().getComWellsfargoEpdUpoAvroPmtInf().getCdtrAgt().getComWellsfargoEpdUpoAvroCdtrAgt().getFinInstnId2().getComWellsfargoEpdUpoAvroFinInstnId2().getBicfi().equals("WFBIUS1A")
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/listener/service/intermediator/IntermediatorService.java:23]: Try to reuse the code, use a variable.

File: src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java [hunk 4]
```diff
@@ -100,15 +97,19 @@ public boolean saveFileToDisk(String directoryPath, String fileName, String resp
...
-    public Optional<Upo> getUPO(String directoryPath, String fileName) {
+    public Optional<UpoDto> getUPO(String directoryPath, String fileName) {
         ClassLoader classloader = Thread.currentThread().getContextClassLoader();
-        try (InputStream is = classloader.getResourceAsStream(directoryPath + fileName);) {
-            Upo upo = new ObjectMapper().readValue(is, Upo.class);
+        try (InputStream is = classloader.getResourceAsStream(directoryPath + fileName)) {
+            UpoDto upo = new ObjectMapper().readValue(is, UpoDto.class);
             return Optional.of(upo);
         } catch (IOException e) {
             e.printStackTrace();
```
Comment (rkpandita) [inline src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java:106]: Use a logger

File: src/test/java/com/wf/stp/rmbs/unit/RuleBasedMessageSanitizerApplicationUnitTests.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    throw new RuntimeException(e);
+                }
+                UpoDto expectedUpo = getUPO("unit_test_cases/service_router/expected_op/", fileName).get();
+                boolean isReponseMatching = false;
+                try {
+                    String strExpected = new ObjectMapper().writeValueAsString(expectedUpo);
+                    String strResponse = new ObjectMapper().writeValueAsString(response);
+                    isReponseMatching = strExpected.equalsIgnoreCase(strResponse);
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
```
Comment (rkpandita) [inline src/test/java/com/wf/stp/rmbs/unit/RuleBasedMessageSanitizerApplicationUnitTests.java:60]: Use a logger

File: src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java [hunk 6]
```diff
@@ -24,38 +25,34 @@
...
             if (!path.toFile().isDirectory()) {
                 String fileName = path.getFileName().toString();
                 log.info("Filename:" + fileName);
-                Upo inputUpo = getUPO("scenarios/positive/input/", fileName).get();
+                UpoDto inputUpo = getUPO("scenarios/positive/input/", fileName).get();
                 log.info("i/p upo:" + inputUpo);
-                Upo response = null;
+                UpoDto response = null;
                 try {
                     response = readAPIRepsonse(new ObjectMapper().writeValueAsString(inputUpo)).get();
```
Comment (rkpandita) [inline src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java:51]: Please spell the Method name as readAPIResponse()

File: src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java [hunk 7]
```diff
@@ -24,38 +25,34 @@
...
     private MockMvc mvc;
 
-    public static String readFileAsString(String file) throws Exception {
-        return new String(Files.readAllBytes(Paths.get(file)));
-    }
-
     @Test
     @DisplayName("Positive test cases")
-    void Positive_CheckTransactionID() throws Exception, JsonProcessingException {
+    void Positive_CheckTransactionID() throws IOException {
```
[the remark this answers is not in the document]
Comment (monil-persistent) [reply src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java]: Will be discussing this in the future.

File: src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java [hunk 8]
```diff
@@ -39,7 +39,7 @@ class RuleBasedMessageSanitizerApplicationIntgTests {
 
     @Test
     @DisplayName("Positive test cases")
-    void Positive_CheckTransactionID() throws IOException {
+    void positive_CheckTransactionID() throws IOException {
```
Comment (rkpandita) [inline src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java:42]: Method name can be checkTransactionId_Positive()

Comment (monil-persistent) [reply src/test/java/com/wf/stp/rmbs/intg/RuleBasedMessageSanitizerApplicationIntgTests.java:42]: Yogesh's Code, will discuss this later.

File: src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java [hunk 9]
```diff
@@ -13,16 +13,15 @@
 @ControllerAdvice
 public class RbmsExceptionController {
     @ExceptionHandler(value = RbmsException.class)
-    public ResponseEntity<Object> exception(RbmsException exception) {
-//        log.info("Field is null", exception);
-        Map<String, Object> map = new HashMap<String, Object>();
-        map.put("message", exception.getErrorMsg());
-        map.put("status", exception.getErrorCode());
-        return new ResponseEntity<>(map, HttpStatus.INTERNAL_SERVER_ERROR);
+    public ResponseEntity<Object> exception(RbmsException ex) {
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/exception/RbmsExceptionController.java:16]: Method name should be a verb eg. 'handleException()'

**Review by rkpandita — APPROVED** [review]
Looks good to me.

**Review by YogeshPitale — APPROVED** [review]
Looks good

**Pull request merged**